oscarapi/apps/basket/utils.py

```diff
@@ -0,0 +1,45 @@
+from oscar.core.loading import get_model, get_class
+
+
+__all__ = ('prepare_basket', 'get_basket', 'apply_offers', )
+
+
+Applicator = get_class('offer.utils', 'Applicator')
+Selector = None
+
+
+def apply_offers(request, basket):
+    "Apply offers and discounts to cart"
+    if not basket.is_empty:
+        Applicator().apply(request, basket)
+
+
+def prepare_basket(basket, request):
+    # fixes too early import of Selector
+    global Selector
```

Comment:
@maerteijn you know all about this issue, that I so elegantly solved with the use of global variables right?